irepela/homework_10/generators.py

```diff
@@ -0,0 +1,43 @@
+
+def simple_generator(iterable):
+    index = 0
+    while index < len(iterable):
+        yield iterable[index]
+        index += 1
+
+
+def reversed_generator(iterable):
+    index = len(iterable) - 1
+    while index >= 0:
+        yield iterable[index]
+        index -= 1
+
+
+def cycle_generator(iterable):
+    index = 0
+    while True:
+        if index >= len(iterable):
+            index = 0
+
+        try:
+            yield iterable[index]
+            index += 1
+        except IndexError:
+            break
```

Comment:
why would you expect  Index error if you check     index >= len(iterable) and reset cursor

Reply:
Fixed